lib_device_control/api/control_shared.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+#ifndef __control_shared_h__
+#define __control_shared_h__
+
+#include <stdint.h>
+#include <stddef.h>
+
+/**
+ * \defgroup control_shared
+ *
+ * The shared API for using the device control library on the device and host
+ * @{
+ */
+
+/**
+ * This is the version of control protocol. Used to check compatibility
+ */
+#define CONTROL_VERSION 0x10
+
+/**
+ * @{
+ * These types are used in control functions to identify the resource id,
+ * command, version, and status.
+ */
+typedef uint8_t control_resid_t;
+typedef uint8_t control_cmd_t;
+typedef uint8_t control_version_t;
+typedef uint8_t control_status_t;
+typedef uint8_t control_ret_t;
+/**@}*/
+
+/**
+ * This type enumerates the possible outcomes from a control transaction.
+ *
+ */
+/* TODO: Some of the enum values below should be used

```

Comment:
This needs fixing or describe this issue here - a link to an internal system should not make it into external code.

Reply:
I agree. The comment and the extra enum values below will be removed if this PR is merged.